src/components/organisms/test/hooks/use-test.ts

```diff
@@ -1,82 +1,86 @@
-import { FormEvent, useEffect, useState } from 'react';
-import { useAppSelector } from 'store';
-import { selectIsTestLoading, selectTest } from 'store/test/selectors';
-import { Controls } from 'utils/constants';
-import type { TestModel } from 'api/lessons';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { calculatePercent } from 'utils/calculate-percent';
+import { AVERAGE_TEST_RESULT, LOADING_PROCESS_MAP } from 'utils/constants';
+import { useAppDispatch, useAppSelector } from 'store';
+import {
+  selectAnswersOnValidate,
+  selectProcessCreationTest,
+  selectTest,
+  selectTestProcess,
+  selectTestResult,
+} from 'store/test/selectors';
+import { updateAnswerReset } from 'store/test/slice';
+import { createTest, validateTest } from 'store/test/thunk';
 
 /**
  * Хук реализует логику прохождения теста.
  * Возвращает объект данных и обработчков для отображения их в интерфейсе.
  *
- * @returns \{ isSubmitted, isSuccess, isLoading, testResultPercent, test, onSubmit, handleButtonDisabledState, retake \}
+ * @returns \{ isSubmitted, isSuccess, isLoading, testResultPercent, test, onSubmit, handleButtonDisabledState, retake, createNewTest \}
  * */
 
 export const useTest = () => {
+  const { lessonId } = useParams();
   const [isSubmitted, setIsSubmitted] = useState(false);
   const [isSuccess, setIsSuccess] = useState(false);
   const [testResultPercent, setTestResultPercent] = useState(0);
 
-  // TODO удалить типы после типизации стора
-  // TODO https://github.com/Studio-Yandex-Practicum/lizaalert_frontend/issues/397
-  const test = useAppSelector<TestModel>(selectTest);
-  const isLoading = useAppSelector<boolean>(selectIsTestLoading);
+  const test = useAppSelector(selectTest);
+  const answers = useAppSelector(selectAnswersOnValidate);
+  const testProcess = useAppSelector(selectTestProcess);
+  const isTestCreated = useAppSelector(selectProcessCreationTest);
```

Comment:
В `isTestCreated` лежит не булева, а ProcessEnum, именование не очень подходит:

```suggestion
  const testCreationProcess = useAppSelector(selectProcessCreationTest);
```

Либо сразу сравнить, чтобы была булева:
```tsx
const isTestCreated = useAppSelector(selectProcessCreationTest) === ProcessEnum.Failed;
```